### PR TITLE
ASM-5883 Firmware updates failing when force_restart not selected

### PR DESCRIPTION
**Pull request by logankimmel**
We need to monitor the end-state statuses on an individual basis.
Direct update components will finish with "Completed" while staged
update components will finish with "Scheduled" when we wait for
a user to manually perform the restart

### CODE REVIEW

**Comment by logankimmel** [comment]
marked as DNM, waiting on some discovery

**Comment by logankimmel** [comment]
Here are the pertinent files showing successful run: https://gist.github.com/logankimmel/cf4bb3ac3bbfb7a882ba

@gavin-scott This is ready for merge

File: lib/puppet/provider/idrac_fw_installfromuri/wsman.rb [hunk 1]
```diff
@@ -85,26 +85,42 @@ def update(firmware_list)
         Puppet.debug("Firmware downloaded to idrac")
       end
     end
+    #
+    # # Reboot if necessary
+    # reboot_job_ids = statuses.each.values.map do |v|
+    #   v[:job_id] unless NO_REBOOT_COMPONENT_IDS.include?(v[:firmware]['component_id'].to_i)
+    # end.compact
```
Comment (gavin-scott) [inline lib/puppet/provider/idrac_fw_installfromuri/wsman.rb]: Can you remove the commented-out code?

File: lib/puppet/provider/idrac_fw_installfromuri/wsman.rb [hunk 2]
```diff
@@ -85,26 +85,42 @@ def update(firmware_list)
...
-    if reboot_required
+    #
+    # if reboot_job_ids.empty?
+    #   Puppet.debug("Reboot not required")
+    #   reboot_required = false
+    #   update_complete = 'Completed'
+    # else
+    #   reboot_required = true
+    #   update_complete = @force_restart ? 'Completed' : 'Scheduled'
+    # end
```
Comment (gavin-scott) [inline lib/puppet/provider/idrac_fw_installfromuri/wsman.rb]: Remove commented out code.

**Comment by logankimmel** [comment]
@gavin-scott my mistake.  removed commented out code

File: lib/puppet/provider/idrac_fw_installfromuri/wsman.rb [hunk 3]
```diff
@@ -117,17 +133,20 @@ def update(firmware_list)
...
-    # Poll for all jobs to complete or time out
-    until statuses.values.all? { |v| v[:status] =~ /#{update_complete}|Failed|InternalTimeout/ }
-      statuses.each do |key, val|
-        if Time.now - val[:start_time] > MAX_WAIT_SECONDS
-          val[:status] = 'InternalTimeout'
-        else
-          val[:status] = checkjobstatus key
-          Puppet.debug("Job Status #{key}: #{val[:status]}")
+    [scheduled_endstate_firmwares, completed_endstate_firmwares].each do |firmware_set|
+      unless firmware_set.empty?
```
Comment (gavin-scott) [inline lib/puppet/provider/idrac_fw_installfromuri/wsman.rb]: I don't think this line is needed. `.all?` should return true for an empty list:

    [1] pry(main)> [].all?(&:even?)
    => true

**Comment by jkrauskopf** [comment]
@logankimmel Is this ready for review again?

**Pull request merged**